src/tasks/MonoTargetsTasks/MarshalingPInvokeScanner/MarshalingPInvokeScanner.cs

```diff
@@ -0,0 +1,333 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Reflection;
+using System.Reflection.PortableExecutable;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+internal enum Compatibility
+{
+    Compatible,
+    Incompatible,
+    Inconclusive
+}
+
+internal sealed class InconclusiveCompatibilityCollection
+{
+    private Dictionary<string, HashSet<string>> _data = new();
+
+    public bool IsEmpty => _data.Count == 0;
+
+    public void Add(string assyName, string namespaceName, string typeName)
+    {
+        HashSet<string>? incAssyTypes;
+
+        if(!_data.TryGetValue(assyName, out incAssyTypes))
+        {
+            incAssyTypes = new();
+            _data.Add(assyName, incAssyTypes);
+        }
+
+        incAssyTypes.Add(string.Join(":", namespaceName, typeName));
```

Comment:
super nit:
```suggestion
        incAssyTypes.Add($"{namespaceName}:{typeName}");
```
.. just simpler to read.